R/table-name.R

```diff
@@ -0,0 +1,189 @@
+# table source = table id or sql
+
+# table id =
+# * interface to outside world; many ways to specify
+# * always refers to exactly one table
+# * but all converted to table name ASAP
+
+# table name
+# * escaped string
+# * internal (and backend) use only; not user facing
+# * can be vector of multiple names
+# * object names are always assumed to be table names
+
+# table_name --------------------------------------------------------------
+
+table_name <- function(x) {
```

Comment:
`table_name` is a bit confusing to me. This is more like a table identifier and only the last bit would be the name.

Reply:
Yeah, I struggled to come up with good names for these functions 😞 especially while avoiding the previous names. What do you think of table path?